R/output.R

```diff
@@ -157,5 +169,3 @@ stop_bad_handlers <- function() {
     call = call("new_output_handler")
   ))
 }
-
-default_output_handler <- new_output_handler()
```

Comment:
First, **knitr** uses this internal function name
https://github.com/yihui/knitr/blob/aa34f1b85f6bb899b1e23a72ebd64054afb3c695/R/utils.R#L860-L861

So removing it breaking knitr. I did not review this PR so did not test it. Only tried dev evaluate today. 

Moreover, this was added some years ago on purpose for performance reason it seems 
https://github.com/r-lib/evaluate/commit/78059bb98ff7fdbb0e281ad8b375a9e7fc96bcab
Commit message
> for a large number of expressions, evaluating new_output_handler() again and again can take some time and also unnecessary, so just create a default copy beforehand and use it

How should we proceed regarding **knitr** ? Can we revert this so that **knitr** is not impacted ? Or should we make **knitr** do differently ? 

But this will mean that older knitr won't work with newer evaluate. Which is already the case IMO as I needed already to do two commits in **knitr** to prepare for new evaluate  (5510d3f8a7e0beb1356330a67892e2eeb2dd1ef2 and f4d670ffa81cd330b564077e7f98f8f41d2876a8)

Reply:
If knitr uses it, I can bring it back (but compute it in a safer way)